client/cli/src/config.rs

```diff
@@ -248,6 +252,17 @@ pub trait CliConfiguration<DCV: DefaultConfigurationValues = ()>: Sized {
 			.unwrap_or_else(|| Ok(Default::default()))
 	}
 
+	/// Get the delayed canonicalization mode.
+	///
+	/// By default this is retrieved from `delayed_canonicalization` if it is available.
+	/// Otherwise the mode is active.
+	fn delayed_canonicalization(&self) -> Result<Option<u32>> {
```

Comment:
looks like you could have `Result<u32>` here instead or why always return `Some(DELAYED_CANONICALIZATION)` if pruning_params() is None?

Reply:
ok it looks to be part of the CliConfiguration perhaps for a uniform API or something...

Reply:
I do remember thinking the same and being a bit confused by the CliConfiguration API.
The reasoning behind it, I think is for consistency and I think of it the following way:

- The `delay_canonicalization` option could also be disabled if stated explicitly which corresponds to the `Option<u32>` part. While the `Result` part is to keep the API consistent with other methods (ie `trie_cache_maximum_size` and the rest) 

 - The `unwrap_or` from above is because  `self.pruning_params()` returns `Option<&PruningParams>` and when we map it to obtain the `delayed_canonicalization` that would turn into `Option<Option<u32>>`. The first layer is stripped with the `unwrap_or` similar to other `state_pruning` and `blocks_pruning`.